video/out/opengl/user_shaders.h

```diff
@@ -68,10 +73,32 @@ struct gl_user_shader {
     int components;
 };
 
-// Parse the next shader pass from `body`. The `struct bstr` is modified by the
+struct gl_user_shader_tex {
+    struct bstr name;
+    int w, h, d;
+    int components;
+    int bytes;
+    int mpgl_type;
+    GLenum gl_target;
+    GLenum gl_filter;
+    GLenum gl_border;
+    void *texdata;
+    // for video.c
+    GLenum gl_tex;
+};
+
+struct gl_user_shader {
+    enum user_shader_type type;
+    union {
+        struct gl_user_shader_hook hook;
+        struct gl_user_shader_tex tex;
+    } v;
```

Comment:
I would argue that tagged unions are a concept that *only* exists in C. (Other languages have real disjoint sum types)

So you'd be happy if I just made it have both structs next to each other, with the attached semantic of `only one is relevant`? I think at that point I'd rather go with the scott encoding idea.